rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -161,19 +163,71 @@ rmw_fastrtps_shared_cpp::create_participant(
   eprosima::fastdds::dds::DomainParticipantQos domainParticipantQos =
     eprosima::fastdds::dds::DomainParticipantFactory::get_instance()->get_default_participant_qos();
 
+  // Configure discovery
+  switch (discovery_options->automatic_discovery_range) {
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_NOT_SET:
+      RMW_SET_ERROR_MSG("automatic discovery range must be set");
+      return nullptr;
+      break;
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_OFF:
+      domainParticipantQos.wire_protocol().builtin.discovery_config.discoveryProtocol =
+        eprosima::fastrtps::rtps::DiscoveryProtocol_t::NONE;
+      break;
+    case RMW_AUTOMATIC_DISCOVERY_RANGE_LOCALHOST: {
+        // Clear the list of multicast listening locators
+        domainParticipantQos.wire_protocol().builtin.metatrafficMulticastLocatorList.clear();
+        // Add a unicast locator to prevent creation of default multicast locator
+        eprosima::fastrtps::rtps::Locator_t default_unicast_locator;
+        domainParticipantQos.wire_protocol()
+        .builtin.metatrafficUnicastLocatorList.push_back(default_unicast_locator);
+        // Add a shared memory transport
```

Comment:
Note: It is recommended to disable the built-in transports when you don't want them to be instantiated (like in this case)

```suggestion
        // Disabling built-in transports, since we are configuring our own.
        domainParticipantQos.transport().use_builtin_transports = false;
        // Add a shared memory transport
```

Reply:
Disabled built-in transports in d1e94da